modules/apps/frontend-js/frontend-js-clay-sample-web/src/main/java/com/liferay/frontend/js/clay/sample/web/internal/portlet/FrontendJSClaySampleWebPortlet.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.js.clay.sample.web.internal.portlet;
+
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCPortlet;
+
+import javax.portlet.Portlet;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Bryce Osterhaus
+ */
+@Component(
```

Comment:
@jbalsas Isn't this a good use case for those pure JS portlets we support?

Reply:
It could be, but that would also be an additional friction point or something we might need to debug if things didn't go as expected... so for the purpose of this experiment, the closer to how code builds and runs in DXP the better... I guess... 🤔

Reply:
Yeah, I guess. It just made sense to avoid this Java boilerplate that we don't necessarily need right now. Plus, for what I understand the purpose of this portlet, we probably are never gonna need it. But I haven't played with the pure JS portlets stuff so I don't know what kind of problems we might run into.

![tenor-176079883](https://user-images.githubusercontent.com/156388/89307157-1fa2b380-d647-11ea-8459-7a0f46e76252.gif)

But I'm all for avoiding friction

![tenor-31438121](https://user-images.githubusercontent.com/156388/89307274-4234cc80-d647-11ea-8dd5-01716f29e91b.gif)

Reply:
Love the poor little pigglet 😂